### ISSUE TITLE
Failure due to http://metadata.ftp-master.debian.org/changelogs/ being incomplete

### ISSUE THREAD

**Issue by boltronics** [systemd/mkosi#1030]
My Debian bullseye image builds are failing with:

```
‣  Recording packages in manifest…
E: Failed to fetch http://metadata.ftp-master.debian.org/changelogs/main/d/dpkg/dpkg_1.20.10_changelog  Changelog unavailable for dpkg=1.20.10 (404  Not Found [IP: 151.101.82.132 80])
‣ Error: Workspace command apt-get --quiet --quiet changelog dpkg --option Acquire::Changelogs::URI::Override::Origin::Debian=http://metadata.ftp-master.debian.org/changelogs/@CHANGEPATH@_changelog returned non-zero exit code 100.
‣  (Unmounting image)
```

![image](https://user-images.githubusercontent.com/250531/177036026-c28af79d-1dd5-4ea2-828e-1b3fbe6b0ecd.png)

It's probably missing because this was a security fix.

```
$ apt-cache policy dpkg
dpkg:
  Installed: 1.20.10
  Candidate: 1.20.10
  Version table:
 *** 1.20.10 500
        500 http://security.debian.org/debian-security bullseye-security/main amd64 Packages
        100 /var/lib/dpkg/status
     1.20.9 500
        500 http://cyclops.monsters:3142/debian bullseye/main amd64 Packages
$
```

This occurred because I had `ManifestFormat=changelog` set, and `WithDocs=` was unspecified (defaulting to "yes" in this case it would seem).

Personally, I'm not a fan of external dependencies being required, as that's going to be fragile - specially when it's not obvious to the user that it's happening (until it breaks). My suggestion would be to have `WithDocs=` default to `yes` if `ManifestFormat=changelog` is used, and perhaps to have a note regarding this behaviour in both sections of the man page.

**Comment by behrmann** [comment]
I'm not quite sure I understand the problem.

`WithDocs=` always defaults to false, but it rather works the other way around: If `WithDocs=` is false, we'll clean up the docs after they are installed. I don't think apt has a robust mechanism to exclude docs from being installed in the first place. Docs packages might be recommends, but `--no-install-recommends` is an all or nothing proposition and while I think, that too often recommends are everything and the kitchen sink (what likely should be suggests), not installing recommends and just pulling in packages one actually wants can be extremely brittle because packages sometimes change their names and or dependencies between releases. One might also exclude the documentation paths with dpkg, but that won't save us from pulling in dependencies.

Long story short: If the mirror you are installing from is broken, I don't think that's something we can solve. 

Happy to help, if I misunderstood your problem, though. Maybe you could restate it or add some more information.

**Comment by boltronics** [comment]
You're right, my apologies. I made an assumption about `WithDocs=` which was incorrect. Indeed, setting `WithDoc=no` in the Content section doesn't make any difference, and the error still occurs.

The only work-around then would seem to be to not set `ManifestFormat=changelog` (which is what I ended up doing initially since I'm fine with this).

I don't think the mirror is broken (are there even mirrors for this - I think it's a single source + CDN?). Tested from Australia and the USA (west coast) via VPN. I think the metadata simply doesn't exist for these kinds of packages, which `mkosi` incorrectly assumes will always exist. Happy to be proven wrong though, if there are other mirrors somewhere that have this file.

**Comment by bluca** [comment]
That was a DSA upload (security queue) so it doesn't go through the main mirrors, but the security file server - not sure if it publishes the changelogs

**Comment by bluca** [comment]
They will eventually show up there, at the next stable release, see: https://bugs.debian.org/490848

**Comment by DaanDeMeyer** [comment]
I think this can be closed? Let us know if this is still a problem
